File: 로그파일재정렬.py
#문자열 앞 부분은 식별자.
# 문자 로그보다 숫자 로그가 앞에 온다.
# 식벼자는 순서영향이 없으나 문자가 동일하면 식별자자 순으로 한다.True
# 순자 로그는 입력 순서대로 한다.

test = ["digl 8 1 5 1", "let1 art can", "dig2 3 6", "let2 own kit dig", "let3 art zero"]

def main(test):
    int_loglist = []
    str_loglist =[]

    for i in test:
        temp_list = list(i.split())
        del temp_list[0]
        temp_str = ''

        for j in temp_list:
          temp_str += str(j)
        
        if temp_str.isdecimal() == True:
            int_loglist.append(i)
        else:
            str_loglist.append(i)

    int_loglist.sort()
    str_loglist.sort()
    str_loglist += int_loglist

    return str_loglist      

print(main(test))

## 답안

def solution(test):
    letters, digits = []
    for log in test:
        if log.split()[1].isdigit():
            digits.append(log)
        else:
            letters.append(log)
    letters.sort(key=lambda x: (x.split()[1:], x.split()[0]))
    return letters + digits

File: go/로그파일재정렬/로그파일재정렬.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

/* 로그의 가장 앞 부분은 식별자
문자로 구성된 로그가 숫자 로그보다 앞에 온다.
식별자는 순서 영향에 끼치지 않는다. 문자가 동일할 경우 식별자 순으로 한다.
숫자 로그는 입력 순서대로 한다. */

func dataCleansing (testCase []string) []string {
  cleanLogs := make([]string, 0)

	for i := 0 ; i < len(testCase); i++ {
		sliceStr := strings.Split(testCase[i], " ")
		completStr := sliceStr[1:]
		cleanLogs = append(cleanLogs, completStr[0])
	}
		return cleanLogs
} 

func changeStringInt(str string) string{
	var result = "string"
	slice := strings.Split(str, " ")

	switch slice[0] {
 		case "1", "2", "3", "4", "5", "6", "7", "8", "9", "0":
			result = "integer"
	 default:
		result = "string"
 }

	return result
}

func CheckString (testCase []string, cleanLogs []string) []string{
	var strLogs = make([]string, 0)
	var intLogs = make([]string, 0)

	for i := 0 ; i < len(cleanLogs); i++ {

		if changeStringInt(cleanLogs[i]) == "string" {
			strLogs = append(strLogs, testCase[i])
		} else {
			intLogs = append(intLogs, testCase[i])
		}	
	}

	sort.Strings(strLogs)	
	return append(strLogs, intLogs...)
}


func main(){
	var testCase = []string{"dig1 8 1 5 1", "let1 art can", "dig2 3 6", "let2 own kit dig", "let3 art zero"}
	//result := CheckString(testCase, dataCleansing(testCase))

	//fmt.Print(result)
	fmt.Printf("%#v\n", testCase)
}

